editor/src/clj/editor/render_target.clj

```diff
@@ -0,0 +1,129 @@
+
+(g/defnk produce-pb-msg
+  [color-attachments depth-stencil-attachment-width depth-stencil-attachment-height depth-stencil-attachment-format depth-stencil-attachment-texture-storage]
+  {:color-attachments color-attachments
+   :depth-stencil-attachment {:width depth-stencil-attachment-width
+                              :height depth-stencil-attachment-height
+                              :format depth-stencil-attachment-format
+                              :texture-storage depth-stencil-attachment-texture-storage}})
+
+(defn build-render-target
+  [resource dep-resources user-data]
+  {:resource resource
+   :content (protobuf/map->bytes RenderTarget$RenderTargetDesc (:pb-msg user-data))})
+
+(g/defnk produce-build-targets
+  [_node-id resource pb-msg]
+  [(bt/with-content-hash
+     {:node-id _node-id
+      :resource (workspace/make-build-resource resource)
+      :build-fn build-render-target
+      :user-data {:pb-msg pb-msg}})])
+
+(defn- generate-gpu-texture [args request-id params unit]
+  (texture/image-texture request-id nil))
+
+(g/defnode RenderTargetNode
+  (inherits resource-node/ResourceNode)
+
+  (property color-attachments g/Any (dynamic visible (g/constantly false)))
+  (property depth-stencil-attachment-width g/Num (dynamic visible (g/constantly false)))
+  (property depth-stencil-attachment-height g/Num (dynamic visible (g/constantly false)))
+  (property depth-stencil-attachment-format g/Any (dynamic visible (g/constantly false)))
+  (property depth-stencil-attachment-texture-storage g/Bool (dynamic visible (g/constantly false)))
+
+  (output pb-msg g/Any :cached produce-pb-msg)
+  (output save-value g/Any (gu/passthrough pb-msg))
+  (output form-data g/Any produce-form-data)
+  (output gpu-texture-generator g/Any {:f generate-gpu-texture })
+  (output build-targets g/Any :cached produce-build-targets))
+
+(defn load-render-target [project self resource render-target]
+  (let [depth-stencil-attachment (:depth-stencil-attachment render-target)]
+    (concat
+      (g/set-property self :color-attachments (:color-attachments render-target))
+      (g/set-property self :depth-stencil-attachment-width (:width depth-stencil-attachment))
+      (g/set-property self :depth-stencil-attachment-height (:height depth-stencil-attachment))
+      (g/set-property self :depth-stencil-attachment-format (:format depth-stencil-attachment))
+      (g/set-property self :depth-stencil-attachment-texture-storage (:texture-storage depth-stencil-attachment)))))
+
+(defn register-resource-types
+  [workspace]
+  (resource-node/register-ddf-resource-type workspace
+                                            :ext "render_target"
+                                            :node-type RenderTargetNode
+                                            :ddf-type RenderTarget$RenderTargetDesc
+                                            :load-fn load-render-target
+                                            :icon atlas-icon
```

Comment:
Should we really use the `atlas-icon` for render targets? `icons/32/Icons_36-Texture.png` seems like it might be more suitable?